pkg/kapp/app/labeled_app.go

```diff
@@ -59,11 +59,13 @@ func (a *LabeledApp) Delete() error {
 	}
 
 	if len(rs) > 0 {
-		var resourceNames []string
-		for _, res := range rs {
-			resourceNames = append(resourceNames, res.Description())
+		if checkResource {
```

Comment:
How about we return nil if `!checkResource` after line number 54 itself so that we would save on the list call as well, it's only required for checking if all resources are gone.

Reply:
Good idea!